libraries/AP_Compass/AP_Compass_QMC5883P.h

```diff
@@ -0,0 +1,68 @@
+/*
+ * This file is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This file is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Driver by Lokesh Ramina, Jan 2022
+ */
+#pragma once
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_HAL/I2CDevice.h>
+#include <AP_Math/AP_Math.h>
+
+
+#include "AP_Compass.h"
+#include "AP_Compass_Backend.h"
+
+#ifndef HAL_COMPASS_QMC5883P_I2C_ADDR
+#define HAL_COMPASS_QMC5883P_I2C_ADDR 0x2C
+#endif
+
+/*
+  setup default orientations
+ */
+#ifndef HAL_COMPASS_QMC5883P_ORIENTATION_EXTERNAL
+#define HAL_COMPASS_QMC5883P_ORIENTATION_EXTERNAL ROTATION_ROLL_180
+#endif
+
+#ifndef HAL_COMPASS_QMC5883P_ORIENTATION_INTERNAL
+#define HAL_COMPASS_QMC5883P_ORIENTATION_INTERNAL ROTATION_ROLL_180_YAW_270
+#endif
```

Comment:
Maybe these defaults should be ROTATION_NONE? I'm not sure. 

```suggestion
#ifndef HAL_COMPASS_QMC5883P_ORIENTATION_EXTERNAL
#define HAL_COMPASS_QMC5883P_ORIENTATION_EXTERNAL ROTATION_ROLL_180
#endif

#ifndef HAL_COMPASS_QMC5883P_ORIENTATION_INTERNAL
#define HAL_COMPASS_QMC5883P_ORIENTATION_INTERNAL ROTATION_ROLL_180_YAW_270
#endif
```

Reply:
I am going to test it today as the board wasn't working over weekend.

Reply:
> LGTM once you test again with your board due to the number of changes and rebases :)

The Compass is working properly after the code rebase.